src/tree/ResolvableTreeItemBase.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext, nonNullProp, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { ThemeIcon } from "vscode";
+import { AppResource, AppResourceResolver, GroupableResource, GroupingConfig, ResolvedAppResourceTreeItemBase } from "../api";
+import { applicationResourceResolvers } from "../api/registerApplicationResourceResolver";
+import { getAzureExtensions } from "../AzExtWrapper";
+import { ext } from "../extensionVariables";
+import { InstallableResourceTreeItem } from "./InstallableResourceTreeItem";
+
+export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem implements GroupableResource {
+
+    public groupConfig: GroupingConfig;
+    public resolveResult: ResolvedAppResourceTreeItemBase | undefined | null;
+    public data: AppResource;
+
+    public async loadMoreChildrenImpl(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        await this.resolve(clearCache, context);
+        if (this.resolveResult && this.resolveResult.loadMoreChildrenImpl) {
+            return await this.resolveResult.loadMoreChildrenImpl(clearCache, context);
+        } else {
+            return [];
+        }
+    }
+
+    public hasMoreChildrenImpl(): boolean {
+        return false;
+    }
+
+    public async resolve(clearCache: boolean, context: IActionContext): Promise<void> {
+
+        const resolver = this.getResolver();
+
+        if (!resolver) {
+            this.resolveResult = {
+                id: this.data.id,
+                contextValue: 'noResolver',
+                label: 'No resolver',
+                description: 'No resolver found for this resource',
+                iconPath: new ThemeIcon('error'),
+            }
+        }
+
+        await this.runWithTemporaryDescription(context, 'Resolving...', async () => {
```

Comment:
```suggestion
        await this.runWithTemporaryDescription(context, 'Loading...', async () => {
```

Users won't know what "Resolving" means